src/lib.rs

```diff
@@ -623,6 +648,8 @@ where
   pp_digest: E1::Scalar,
   vk_primary: S1::VerifierKey,
   vk_secondary: S2::VerifierKey,
+  vk_primary_commitment_key: CommitmentKey<E1>,
```

Comment:
I missed this earlier. This will make the verifier key non-succinct. Here's a way to fix this.

We should add a type called `DerandKey` (analogous to `CommitmentKey`) in `CommitmentEngine`:

`  /// Holds the type of the derandomization key
  type DerandKey: Clone + Debug + Send + Sync + Serialize + for<'de> Deserialize<'de>;`

We can then add a method like `derand_key` to `CommitmentEngine` and this can be implemented as follows:

`fn derand_key(ck: &Self::CommitmentKey) -> Self::DerandKey {
    DerandKey { h: ck.h.clone() }
  }
`

Then, the verifier only needs to hold a DerandKey and the `derandomize` method only takes this DerandKey, which is constant-sized, making it succinct.

Does this make sense?

Reply:
This does make sense! Thank you for all of your help and patience with this PR.